model/build/build.go

```diff
@@ -317,3 +331,78 @@ func (b Builds) InsertMany(ctx context.Context, ordered bool) error {
 	_, err := evergreen.GetEnvironment().DB().Collection(Collection).InsertMany(ctx, b.getPayload(), &options.InsertManyOptions{Ordered: &ordered})
 	return errors.Wrap(err, "bulk inserting builds")
 }
+
+// GetFinishedNotificationDescription returns a description of successful/failed tasks for the build,
+// to be used by jobs and notification processing.
+func (b *Build) GetFinishedNotificationDescription(tasks []task.Task) string {
+	success := 0
+	failed := 0
+	systemError := 0
+	other := 0
+	noReport := 0
+	for _, t := range tasks {
+		switch {
+		case t.Status == evergreen.TaskSucceeded:
+			success++
+
+		case t.Status == evergreen.TaskFailed:
+			failed++
+
+		case statusIsSystemFailure(t.Status):
+			systemError++
+
+		case utility.StringSliceContains(evergreen.TaskUncompletedStatuses, t.Status):
+			noReport++
```

Comment:
Should we just return something / error out early here since it shows the build is not finished? 

If we do this, maybe we can use a string -> int map of statuses to count, with `TaskFailed, TaskSucceeded, TaskSystemFailed` being the keys rather than setting up these 5 variables.

Reply:
I didn't write this function (just moved it so it could be used in the units package) but it does look like even if the build isn't finished we still show how many have succeeded / failed so far looking at the description.